Fix flyway CLI install task

DanVanAtta opened the pull request:
- Do not redownload flyway zip file, if zip file already exists
we can skip DL (otherwise fails), and similarly if zip file is
already extracted we can skip the extraction


<!-- 
  Commit comment above summarizing the update.  If multiple commits please 
  summarize the change above. Commit comments should include an overview of
  the updates and the goal and reasoning behind the update.
  Code standards and PR guidelines can be found at:
  - https://github.com/triplea-game/triplea/wiki/Contribution-Guidelines
  - https://github.com/triplea-game/triplea/wiki/Code-Reviews
--> 


## Functional Changes
<!-- Put an X next any that apply -->
[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[] Problem fix:  <!-- Link to bug issue or forum post here -->
[] Other:   <!-- Please specify -->


## Testing
<!--
  Place an X below if applies. Manual testing is a crutch for us, 
  we would prefer to rely on automated testing.
-->

[] Manual testing done

<!-- If manually tested, summarize the testing done below this line. -->


<!-- If there are UI updates, uncomment and include screenshots below -->
<!--
## Screens Shots

### Before

### After
-->


<!-- 
  Uncomment the below and add any additional details that would be helpful for reviewers.
-->
<!--
## Additional Review Notes
-->

RoiEXLab commented on infrastructure/ansible/roles/flyway/tasks/main.yml:
```diff
@@ -36,13 +36,25 @@
+  register: flyway_extraction
+
 - name: extract flyway
+  when: flyway_extraction.stat.exists == False
```
According to the docs of the unarchive module, you could use the `creates` property instead of using a stat.

DanVanAtta replied:
"creates" is a nice new feature, happy to use it here.

RoiEXLab commented on infrastructure/ansible/roles/flyway/tasks/main.yml:
```diff
@@ -36,13 +36,25 @@
+  register: flyway_download
+
 - name: Download flyway CLI
+  when: flyway_download.stat.exists == False
```
Also the default behaviour of the `get_url` module is not to overwrite any existing files according to the docs:
See parameter `force`:
https://docs.ansible.com/ansible/latest/modules/get_url_module.html

DanVanAtta replied:
This update is required, or you get this error:
 "msg": "Destination /tmp/flyway-cli.tar.gz is not writable",

`force` is not ideal as we get a DL and overwrite every time. It turns out if we set a checksum on the get_url task, then it knows enough when to skip download task. I think it's designed this way to have you clean up downloaded files, and/or overwrite them, or set a checksum for them. In any of these cases, if the download fails or is corrupted, then the playbook can be repeated and it'll repair any issues. If you skip download with a corrupt file, then the re-run would not fix the issues.

DanVanAtta commented:
Iterated one more time and fixed what I think are the most salient and remaining problems in the flyway tasks. Previously if we run the role twice it would have error'ed out on the download file. After fixing that, we still had 'changed' reports. I fixed those up, after running the task a second time we get a desirable report of no changes:

![Screenshot from 2019-12-11 21-26-03](https://user-images.githubusercontent.com/12397753/70685038-fb7c0780-1c5c-11ea-93bb-6096a7a1ed7a.png)
